src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotReferSelfInSerializableClassTests.cs

```diff
@@ -0,0 +1,533 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotReferSelfInSerializableClassTests : DiagnosticAnalyzerTestBase
+    {
```

Comment:
Need some test cases (and analyzer code) for arrays and other collections (List<T>, IList<T>, etc), such as:
```csharp
[Serializable]
class TestClass
{
    private TestClass[] testClasses;

    public void TestMethod()
    {
    }
}
```